### PR TITLE
Add mcuboot and SMP support

### PR DESCRIPTION
**Pull request by endian-albin**
Enable image management and DFU when building with bootloader support. Also make this the default.
    
- Remove main/bt threads and enable Bluetooth advertising by default
- Add support for DFU over the Simple Management Protocol
- No automatic image confirmation: use the mcumgr CLI program instead
- Update README
- Check if a watchdog is running and, if so, kick it
- CI: Build mcuboot image and DFU package

Completes #36

### CODE REVIEW

File: app/hypnos/config/bootloader.conf [hunk 1]
```diff
@@ -9,3 +9,36 @@ CONFIG_FLASH=y
 
 # Memory protection
 CONFIG_MPU_ALLOW_FLASH_WRITE=y
+
+# Allow for large Bluetooth data packets.
+CONFIG_BT_L2CAP_TX_MTU=260
```
Comment (sdorre) [inline app/hypnos/config/bootloader.conf:14]: How did you choose this value ?

File: app/west.yml [hunk 2]
```diff
@@ -35,12 +35,21 @@ manifest:
...
     - name: mcuboot
       revision: e88113bbebe34ff2ccc6627ffae885cfeed6fdfd
       path: bootloader/mcuboot
+    - name: mcumgr
+      revision: 5885efb7cabf7b566577b73129c9d277d7d8848d
+      path: modules/lib/mcumgr
     - name: segger
       path: modules/debug/segger
       revision: 6fcf61606d6012d2c44129edc033f59331e268bc
+    - name: tinycbor
```
Comment (sdorre) [inline app/west.yml:50]: why do. you need this library for ?

File: app/hypnos/config/bootloader.conf (hunk 1 above)
Comment (endian-albin) [reply app/hypnos/config/bootloader.conf:14]: I simply copied it from the Zephyr SMP sample application. I've not tried excluding it.

File: app/west.yml (hunk 2 above)
Comment (endian-albin) [reply app/west.yml:50]: It's required by mcumgr. Excluding it leads to:
```
pinetime-hypnos/modules/lib/mcumgr/mgmt/include/mgmt/mgmt.h:24:10: fatal error: tinycbor/cbor.h: No such file or directory
   24 | #include "tinycbor/cbor.h"
```

**Comment by endian-albin** [comment]
Future bootloader support: I've a added a watchdog refresh timer and ISR for compatibility with the [v5.0.0 release candidate](https://github.com/lupyuen/pinetime-rust-mynewt/releases/tag/v5.0.0). It works as expected; without it, the device restarts automatically.

**Comment by endian-albin** [comment]
I've now added Github CI support and made the bootloader/DFU version the new default. To build without bootloader support, just `export BOOTLOADER="off"`.

**Review by sdorre — APPROVED** [review]
Looks good!

Does it work with the  NRF Connect mobile app ?

**Comment by endian-albin** [comment]
> Looks good!
> 
> Does it work with the NRF Connect mobile app ?

Yes, it does.

**Comment by endian-albin** [comment]
I have now successfully 1) flashed the versions of the bootloader and InfiniTime that will be the new factory install, upgraded to Hypnos (using the image built by the CI) via nRF connect, then upgraded again to InfiniTime "develop" via mcumngr CLI . Things seem to work well so I''ll press the merge button.

**Pull request merged**